### PR TITLE
bring back the forgotten tests

### PR DESCRIPTION
**Pull request by trontrytel**
Hi guys. I'm riding the tests wave in scampy. This pull adds the two unit tests that I wrote ages ago: 
 - test equation of state with some randomly generated input
 - tests if grid mean qt == updraft_area * updraft_qt + env_area * env_qt

I also added a wrapper to be able to access cython stuff from python. 

The tolerance for testing the man seems pretty high to me... The test is done after 2 hours simulation time and the relative tolerance is of order 1e-3. Any thoughts?

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -189,6 +189,10 @@
                  runtime_library_dirs=library_dirs)
 extensions.append(_ext)
 
+_ext = Extension('pytest_wrapper', ['pytest_wrapper.pyx'], include_dirs=include_path,
+                 extra_compile_args=extra_compile_args, libraries=libraries, library_dirs=library_dirs,
+                 runtime_library_dirs=library_dirs)
+extensions.append(_ext)
 
 #Build RRTMG
```
Comment (yairchn) [inline setup.py]: you choose to leave the commented part for a reason ?

File: tests/unit/test_eos.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import numpy as np
+
+import pytest
+import pytest_wrapper as wrp
+
+# https://hypothesis.readthedocs.io/en/latest/
+from hypothesis import given, strategies as st
+
+
+@given(p0   = st.floats(min_value = 30000, max_value = 101300), # pressure between 1013hPa - 300hPa
```
Comment (yairchn) [inline tests/unit/test_eos.py]: in deep convection you can get to pressure of 120hPa (18km)

File: tests/unit/test_eos.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    """
+    qv = frac * qt
+
+    pd = wrp.pd_c(p0, qt, qv)
+    pv = wrp.pv_c(p0, qt, qv)
+
+    assert(np.isclose(p0, pd + pv, rtol = 1e-12))
+
+
+@given(p0 = st.floats(min_value = 30000,  max_value = 101300),      # pressure between 1013hPa - 300hPa
```
Comment (yairchn) [inline tests/unit/test_eos.py]: in deep convection you can get to pressure of 120hPa (18km)

File: tests/unit/test_eos.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    # calculate thetali(T) assuming no liquid water or ice are present
+    thetali = wrp.t_to_thetali_c(p0, T, qt, ql, qi)
+    # calculate T(thetali) assuming no liquid water or ice are present
+    T_new = wrp.eos_first_guess_thetal(thetali, pd, pv, qt)
+
+    assert(np.isclose(T, T_new, rtol = 1e-12))
+
+
+@given(z  = st.floats(min_value = 0,     max_value = 11000), # height in the atmosphere around which we will test
```
Comment (yairchn) [inline tests/unit/test_eos.py]: in deep convection you can get to 18km

File: setup.py (hunk 1 above)
Comment (trontrytel) [reply setup.py]: leaving as I found it ;)

**Comment by trontrytel** [comment]
@yairchn  Good point! I remember having issues with the convergence if I was stretching to very low pressures while keeping qt high. But I'll see what I can do for the deep convection limit

File: tests/unit/test_eos.py (hunk 2 above)
Comment (trontrytel) [reply tests/unit/test_eos.py]: 30000 -> 12000

File: tests/unit/test_eos.py (hunk 3 above)
Comment (trontrytel) [reply tests/unit/test_eos.py]: 30000 -> 12000

File: tests/unit/test_eos.py (hunk 4 above)
Comment (trontrytel) [reply tests/unit/test_eos.py]: 11km -> 18km

**Comment by trontrytel** [comment]
@yairchn - I extended the tests to 120hPa and 18km

**Pull request merged**